File: src/main/java/com/skylot222/service/ShipmentService.java
package com.skylot222.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skylot222.domain.Shipment;
import com.skylot222.dto.ShipmentDTO;
import com.skylot222.dto.ShipmentSearchDTO;
import com.skylot222.dto.ShipmentPageDTO;
import com.skylot222.dto.ShipmentConvertCriteriaDTO;
import com.skylot222.service.GenericService;
import com.skylot222.dto.common.RequestDTO;
import com.skylot222.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ShipmentService extends GenericService<Shipment, Integer> {

	List<Shipment> findAll();

	ResultDTO addShipment(ShipmentDTO shipmentDTO, RequestDTO requestDTO);

	ResultDTO updateShipment(ShipmentDTO shipmentDTO, RequestDTO requestDTO);

    Page<Shipment> getAllShipments(Pageable pageable);

    Page<Shipment> getAllShipments(Specification<Shipment> spec, Pageable pageable);

	ResponseEntity<ShipmentPageDTO> getShipments(ShipmentSearchDTO shipmentSearchDTO);
	
	List<ShipmentDTO> convertShipmentsToShipmentDTOs(List<Shipment> shipments, ShipmentConvertCriteriaDTO convertCriteria);

	ShipmentDTO getShipmentDTOById(Integer shipmentId);







}

File: skylot222_ui/src/services/ShipmentService.js
import http from "../http-common"; 

class ShipmentService {
  getAllShipments(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/shipment/shipments`, searchDTO);
  }

  get(shipmentId) {
    return this.getRequest(`/shipment/${shipmentId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/shipment?field=${matchData}`, null);
  }

  addShipment(data) {
    return http.post("/shipment/addShipment", data);
  }

  update(data) {
  	return http.post("/shipment/updateShipment", data);
  }
  
  uploadImage(data,shipmentId) {
  	return http.postForm("/shipment/uploadImage/"+shipmentId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ShipmentService();
